contracts/strategies/idle-tranche/IdleFinanceTrancheStrategy.sol

```diff
@@ -0,0 +1,320 @@
+            if(liquidityGaugeRewardBalance != 0) {
+                IERC20(_liquidityGaugeReward).safeApprove(_universalLiquidator, 0);
+                IERC20(_liquidityGaugeReward).safeApprove(_universalLiquidator, liquidityGaugeRewardBalance);
+
+                ILiquidator(_universalLiquidator).swapTokenOnMultipleDEXes(
+                    liquidityGaugeRewardBalance,
+                    1,
+                    address(this),
+                    storedLiquidationDexes[_liquidityGaugeReward][_rewardToken],
+                    storedLiquidationPaths[_liquidityGaugeReward][_rewardToken]
+                );
+            }
+        }
+        
+        uint256 rewardBalance = IERC20(_rewardToken).balanceOf(address(this));
+        if (!sell() || rewardBalance < sellFloor()) {
+            // Profits can be disabled for possible simplified and rapid exit
+            emit ProfitsNotCollected(sell(), rewardBalance < sellFloor());
+            return;
+        }
+
+        notifyProfitInRewardToken(rewardBalance);
+        
+        uint256 remainingRewardBalance = IERC20(_rewardToken).balanceOf(address(this));
+        if (remainingRewardBalance == 0) {
+        return;
+        }
+
+        IERC20(_rewardToken).safeApprove(_universalLiquidator, 0);
+        IERC20(_rewardToken).safeApprove(_universalLiquidator, remainingRewardBalance);
+
+        // we can accept 1 as minimum because this is called only by a trusted role
+        ILiquidator(_universalLiquidator).swapTokenOnMultipleDEXes(
+            remainingRewardBalance,
+            1,
+            address(this), // target
+            storedLiquidationDexes[_rewardToken][underlying()],
+            storedLiquidationPaths[_rewardToken][underlying()]
+        );
+    }
+
+    /* ========== External ========== */
+
+    function withdrawAllToVault() public restricted {
+        address _rewardPool = rewardPool();
+        address tranche = IIdleCDO(_rewardPool).AATranche();
+        address _liquidityGauge = liquidityGauge();
+
+        _claimRewards();
+        _liquidateReward();
+
+        
+        uint256 liquidityGaugeBalance = ILiquidityGaugeV3(_liquidityGauge).balanceOf(address(this));
+        ILiquidityGaugeV3(_liquidityGauge).withdraw(liquidityGaugeBalance);
+        
+
+        uint256 trancheBalance = IERC20(tranche).balanceOf(address(this));
+        if(trancheBalance != 0) {
+            uint256 redeemed = IIdleCDO(_rewardPool).withdrawAA(trancheBalance);
+
+            if(isSTETH()) {
+                IERC20(STETH).safeApprove(WSTETH, 0);
+                IERC20(STETH).safeApprove(WSTETH, redeemed);
+                IWstETH(WSTETH).wrap(redeemed);
+            }
+        }
+
+        uint256 underlyingBalance = IERC20(underlying()).balanceOf(address(this));
+        IERC20(underlying()).safeTransfer(vault(), underlyingBalance);
+    }
+
+    function withdrawToVault(uint256 amount) external restricted {
```

Comment:
Is there no way to do a partial withdrawal? We do not want to withdraw all to vault if it is not necessary. The funds will sit idle in the vault until the next hardwork